####### GATK-HaplotypeCaller #######
####### running all BAM samples  #######
####### obtaining GVCF per BAM sample #######

# creating a list with bam names
samples=(HG00171 HG00176 HG00185 HG00277 HG00284 HG00290 HG00306 HG00308 HG00318 HG00323 HG00325 HG00335 HG00336 HG00346 HG00349 HG00366 HG00368 HG00378 HG00379 HG00380 HG01048 HG01054 HG01064 HG01069 HG01077 HG01083 HG01089 HG01098 HG01105 HG01107 HG01111 HG01113 HG01134 HG01140 HG01190 HG01253 HG01260 HG01269 HG01312 HG01344 HG01390 HG01405 HG01413 HG01437 HG01440 HG01447 HG01455 HG01459 HG01479 HG01489 HG01556 HG02461 HG02613 HG02614 HG02623 HG02646 HG02648 HG02649 HG02657 HG02661 HG02678 HG02684 HG02690 HG02725 HG02736 HG02771 HG02793 HG02804 HG02811 HG02814 HG02887 HG02947 HG02952 HG02968 HG02970 HG02971 HG02974 HG02983 HG03009 HG03015 HG03019 HG03021 HG03049 HG03063 HG03073 HG03074 HG03077 HG03086 HG03096 HG03097 HG03105 HG03109 HG03114 HG03115 HG03117 HG03124 HG03126 HG03130 HG03139 HG03196 HG03202 HG03209 HG03212 HG03225 HG03229 HG03240 HG03241 HG03298 HG03304 HG03311 HG03380 HG03410 HG03419 HG03445 HG03461 HG03469 HG03473 HG03517 HG03518 HG03577 HG03611 HG03629 HG03705 HG03708 HG03800 HG03805 HG03815 HG03817 HG03830 HG03902 HG03905 HG03922 HG03925 HG04140 HG04161 HG04182 HG04194 NA18954 NA18957 NA18969 NA18971 NA18977 NA18978 NA18980 NA18992 NA18994 NA18998 NA19054 NA19055 NA19057 NA19063 NA19067 NA19078 NA19080 NA19083 NA19701 NA19713 NA19818 NA19901 NA19909 NA19923 NA20276 NA20281 NA20289 NA20317 NA20322 NA20334 NA20336)

# defining paths 
reference="path_to_reference_genome/hs37d5.fas"
bam_dir="..."
out_dir="./01_g.vcf.gz" # path to output directory

# iterating through each bam file
for sample in $samples
do
java -jar gatk-package-4.2.3.0-local.jar HaplotypeCaller -R $reference -I ./$sample*.bam -O $out_dir/$sample.g.vcf.gz -L 12 # -L is the number of chromosome 
done 
